website/versioned_docs/version-0.17.3/getting-started/tutorials/single-page-application.md

```diff
@@ -0,0 +1,1842 @@
+              }
+              Msg::GetProductsSuccess(products) => {
+                  self.state.products = products;
+                  self.state.get_products_loaded = true;
+                  true
+              }
+              Msg::GetProductsError(error) => {
+                  self.state.get_products_error = Some(error);
+                  self.state.get_products_loaded = true;
+                  true
+              }
+-             Msg::AddToCart(product_id) => {
+-                 let product = self
+-                     .state
+-                     .products
+-                     .iter()
+-                     .find(|p: &&Product| p.id == product_id)
+-                     .unwrap();
+-                 let cart_product = self
+-                     .state
+-                     .cart_products
+-                     .iter_mut()
+-                     .find(|cp: &&mut CartProduct| cp.product.id == product_id);
+-                 if let Some(cp) = cart_product {
+-                     cp.quantity += 1;
+-                 } else {
+-                     self.state.cart_products.push(CartProduct {
+-                         product: product.clone(),
+-                         quantity: 1,
+-                     })
+-                 }
+-                 true
+-             }
+          }
+      }
+
+-     fn change(&mut self, _: Self::Properties) -> ShouldRender {
++     fn change(&mut self, props: Self::Properties) -> ShouldRender {
++         self.props = props;
+          true
+      }
+
+      fn view(&self) -> Html {
+          let products: Vec<Html> = self
+              .state
+              .products
+              .iter()
+              .map(|product: &Product| {
+-                 let product_id = product.id;
+                  html! {
+-                   <ProductCard product={product} on_add_to_cart=self.link.callback(move |_| Msg::AddToCart(product_id))/>
++                   <ProductCard product={product} on_add_to_cart=self.props.on_add_to_cart.clone()/>
+                  }
+              })
+              .collect();
+
+-        let cart_value = self
+-            .state
+-            .cart_products
+-            .iter()
+-            .fold(0.0, |acc, cp| acc + (cp.quantity as f64 * cp.product.price));
+
+          if !self.state.get_products_loaded {
+              // No changes
+          } else if let Some(_) = self.state.get_products_error {
+              // No changes
+          } else {
+              html! {
+-               <div>
+-                 <div class="navbar">
+-                     <div class="navbar_title">{"RustMart"}</div>
+-                     <div class="navbar_cart_value">{format!("${:.2}", cart_value)}</div>
+-                 </div>
+                  <div class="product_card_list">{products}</div>
+-               </div>
+              }
+          }
+      }
+  }
+```
+
+Now we can finally add to cart from `ProductDetail` page and we can also see the navbar in all pages

```

Comment:
Not 100% sure what the semantics of this sentence should look like but something like this (depending on the intended meaning of the sentence) might do the trick.
```suggestion
Now we can finally add a link to the "add to cart" component from the `ProductDetail` page and we can also see the navbar on all pages

```